lib/Tool/Authentication.php

```
@@ -122,37 +121,19 @@ protected static function refreshUser(TokenInterface $token, UserProvider $provi
     {
         $user = $token->getUser();
 
-        $userNotFoundByProvider = false;
-        $userClass = $user::class;
-
-        if (!$provider instanceof UserProviderInterface) {
-            throw new \InvalidArgumentException(sprintf('User provider "%s" must implement "%s".', get_debug_type($provider), UserProviderInterface::class));
-        }
-
-        if (!$provider->supportsClass($userClass)) {
+        if (!$provider->supportsClass($user::class)) {
             return null;
         }
 
         try {
-            $refreshedUser = $provider->refreshUser($user);
-            $newToken = clone $token;
-            $newToken->setUser($refreshedUser);
-            $token->setUser($refreshedUser);
+            $token->setUser($provider->refreshUser($user));
 
             return $token;
-        } catch (UnsupportedUserException) {
-            // let's try the next user provider
```

There is no "next" one, as we're not in a loop, so we can just let it throw